coursier/src/main/scala/sbt/librarymanagement/coursier/CoursierDependencyResolution.scala

```diff
@@ -0,0 +1,259 @@
+import sbt.librarymanagement._
+import sbt.util.Logger
+
+case class CoursierModuleDescriptor(
+    directDependencies: Vector[ModuleID],
+    scalaModuleInfo: Option[ScalaModuleInfo],
+    moduleSettings: ModuleSettings,
+    extraInputHash: Long
+) extends ModuleDescriptor
+
+case class CoursierModuleSettings() extends ModuleSettings
+
+private[sbt] class CoursierDependencyResolution(resolvers: Vector[Resolver])
+    extends DependencyResolutionInterface {
+
+  private[coursier] val reorderedResolvers = Resolvers.reorder(resolvers.toSeq)
+
+  /**
+   * Builds a ModuleDescriptor that describes a subproject with dependencies.
+   *
+   * @param moduleSetting It contains the information about the module including the dependencies.
+   * @return A `ModuleDescriptor` describing a subproject and its dependencies.
+   */
+  override def moduleDescriptor(
+      moduleSetting: ModuleDescriptorConfiguration): CoursierModuleDescriptor = {
+    CoursierModuleDescriptor(
+      moduleSetting.dependencies,
+      moduleSetting.scalaModuleInfo,
+      CoursierModuleSettings(),
+      1L // FIXME: use correct value
+    )
+  }
+
+  val ivyHome = sys.props.getOrElse(
+    "ivy.home",
+    new File(sys.props("user.home")).toURI.getPath + ".ivy2"
+  )
+
+  val sbtIvyHome = sys.props.getOrElse(
+    "sbt.ivy.home",
+    ivyHome
+  )
+
+  val ivyProperties = Map(
+    "ivy.home" -> ivyHome,
+    "sbt.ivy.home" -> sbtIvyHome
+  ) ++ sys.props
+
+  /**
+   * Resolves the given module's dependencies performing a retrieval.
+   *
+   * @param module        The module to be resolved.
+   * @param configuration The update configuration.
+   * @param uwconfig      The configuration to handle unresolved warnings.
+   * @param log           The logger.
+   * @return The result, either an unresolved warning or an update report. Note that this
+   *         update report will or will not be successful depending on the `missingOk` option.
+   */
+  override def update(module: ModuleDescriptor,
+                      configuration: UpdateConfiguration,
+                      uwconfig: UnresolvedWarningConfiguration,
+                      log: Logger): Either[UnresolvedWarning, UpdateReport] = {
+
+    if (reorderedResolvers.isEmpty) {
+      log.error(
+        "Dependency resolution is configured with an empty list of resolvers. This is unlikely to work.")
+    }
+
+    val dependencies = module.directDependencies.map(toCoursierDependency).toSet
+    val start = Resolution(dependencies)
+    val authentication = None // TODO: get correct value
+    val ivyConfiguration = ivyProperties // TODO: is it enough?
+    val repositories =
+      reorderedResolvers.flatMap(r => FromSbt.repository(r, ivyConfiguration, log, authentication)) ++ Seq(
+        Cache.ivy2Local,
+        Cache.ivy2Cache)
+
+    import scala.concurrent.ExecutionContext.Implicits.global
+
+    val fetch = Fetch.from(repositories, Cache.fetch[Task](logger = Some(createLogger())))
```

Comment:
`Logger` (`TermDisplay` at least) should be stopped (via `TermDisplay.stop()`). This might result in spurious terminal state else (some progress bar updates not being flushed in particular).

Reply:
done